examples/templates/ecs-container/main.tf

```
@@ -0,0 +1,242 @@
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.28"
+    }
+    coder = {
+      source  = "coder/coder"
+      version = "~> 0.4.9"
+    }
+    cloudinit = {
+      source  = "hashicorp/cloudinit"
+      version = "2.2.0"
+    }
+  }
+}
+
+# required for multi-user data config
+provider "cloudinit" {}
+
+# configure AWS provider with creds present on Coder server host
+provider "aws" {
+  region                   = "us-east-1"
+  shared_config_files      = ["/home/coder/.aws/config"]
+  shared_credentials_files = ["/home/coder/.aws/credentials"]
+}
+
+locals {
+
+  # User data is used to stop/start AWS instances. See:
+  # https://github.com/hashicorp/terraform-provider-aws/issues/22
+
+  user_data_start = <<EOT
+Content-Type: multipart/mixed; boundary="//"
+MIME-Version: 1.0
+
+--//
+Content-Type: text/cloud-config; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="cloud-config.txt"
+
+#cloud-config
+cloud_final_modules:
+- [scripts-user, always]
+hostname: ${lower(data.coder_workspace.me.name)}
+
+--//
+Content-Type: text/x-shellscript; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="userdata.txt"
+
+#!/bin/bash
+sudo -u ubuntu sh -c '${coder_agent.coder.init_script}'
+--//--
+EOT
+
+  user_data_end = <<EOT
+Content-Type: multipart/mixed; boundary="//"
+MIME-Version: 1.0
+--//
+Content-Type: text/cloud-config; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="cloud-config.txt"
+#cloud-config
+cloud_final_modules:
+- [scripts-user, always]
+--//
+Content-Type: text/x-shellscript; charset="us-ascii"
+MIME-Version: 1.0
+Content-Transfer-Encoding: 7bit
+Content-Disposition: attachment; filename="userdata.txt"
+#!/bin/bash
+sudo shutdown -h now
+--//--
+EOT
+}
+
+# user data definitions, for both ECS & Coder
+data "cloudinit_config" "main" {
+  base64_encode = true
+  gzip          = false
+
+  part {
+    content = data.coder_workspace.me.transition == "start" ? local.user_data_start : local.user_data_end
+  }
+  # required for the EC2 instances to join the ECS cluster
+  part {
+    content = "#!/bin/bash\necho ECS_CLUSTER=${aws_ecs_cluster.main.name} >> /etc/ecs/ecs.config"
+  }
+}
+
+# set launch template for EC2 auto-scaling group
+resource "aws_launch_template" "coder-oss-ubuntu" {
+  name_prefix = "coder-oss"
+  # ECS optimized AMI - https://github.com/awsdocs/amazon-ecs-developer-guide/blob/master/doc_source/ecs-optimized_AMI.md
+  image_id      = "ami-03f8a7b55051ae0d4"
+  instance_type = "t2.medium"
+  iam_instance_profile {
+    arn = "arn:aws:iam::816024705881:instance-profile/coder-oss-ecs-role"
+  }
+  user_data = data.cloudinit_config.main.rendered
+}
+
+# provision auto-scaling group to host ECS task definitions
+resource "aws_autoscaling_group" "main" {
+  name                  = "coder-ecs-auto-scaling-group"
```

```suggestion
  name                  = "coder-ecs-${data.coder_workspace.me.id}-auto-scaling-group"
```

Otherwise, only 1 workspace can exist on an AWS account. I ran into this while testing